core/input/gesturehandler.js

```diff
@@ -338,31 +358,76 @@ export default class GestureHandler {
         return true;
     }
 
-    _stateToGesture(state) {
-        switch (state) {
-            case GH_ONETAP:
-                return 'onetap';
-            case GH_TWOTAP:
-                return 'twotap';
-            case GH_THREETAP:
-                return 'threetap';
-            case GH_DRAG:
-                return 'drag';
-            case GH_LONGPRESS:
-                return 'longpress';
-            case GH_TWODRAG:
-                return 'twodrag';
-            case GH_PINCH:
-                return 'pinch';
+    _startLongpressTimeout() {
+        this._stopLongpressTimeout();
+        this._longpressTimeoutId = setTimeout(() => this._longpressTimeout(),
+                                              GH_LONGPRESS_TIMEOUT);
+    }
+
+    _stopLongpressTimeout() {
+        clearTimeout(this._longpressTimeoutId);
+        this._longpressTimeoutId = null;
+    }
+
+    _longpressTimeout() {
+        if (this._hasDetectedGesture()) {
+            Log.Error("A longpress gesture failed, conflict with a different gesture");
```

Comment:
Is this equivalent to an assert? Perhaps we should throw an error in such cases?

Reply:
Yes, that's a good idea